common/beerocks/hostapd/configuration.cpp

```diff
@@ -0,0 +1,23 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "configuration.h"
+
+namespace prplmesh {
+namespace hostapd {
+namespace config {
+
+Configuration::Configuration(const std::string file_name) : m_configuration_file(file_name) {}
```

Comment:
This becomes easier to test if instead of a filename, you give if an `istream&`. That way, you can pass in a stringstream in the test.

Reply:
indeed, and it is possible, and might even be a good idea.
However, from my experience, and since the file format is so delicate, and since the test flow is: load, change, **store**, **compare** - 
- we'll need an an `ostream` for storing (current file name serves both I/O).
- having the ability to load from a string may decrease the chances that one would compare the input to the output.